src/test/java/com/iexec/worker/chain/WorkerpoolAuthorizationServiceTests.java

```diff
@@ -125,6 +129,78 @@ void shouldPurgeAllTasksData() {
+                workerpoolAuthorization.getChainTaskId(),
+                workerpoolAuthorization.getWorkerWallet()
+        );
+
+        final String challenge = ReflectionTestUtils.invokeMethod(
+                workerpoolAuthorizationService,
+                "getChallenge",
+                workerpoolAuthorization);
+
+        assertEquals(expectedChallenge, challenge);
+    }
+    // endregion
+
+    // region isSignedWithEnclaveChallenge
+    @Test
+    void shouldConfirmSignatureIsSignedWithEnclaveChallenge() {
+        final WorkerpoolAuthorization workerpoolAuthorization = getWorkerpoolAuthorization();
+        final Map<String, WorkerpoolAuthorization> workerpoolAuthorizations = new HashMap<>();
+        workerpoolAuthorizations.put(workerpoolAuthorization.getChainTaskId(), workerpoolAuthorization);
+        ReflectionTestUtils.setField(workerpoolAuthorizationService, "workerpoolAuthorizations", workerpoolAuthorizations);
+
+        try (final MockedStatic<SignatureUtils> signatureUtilsMock = mockStatic(SignatureUtils.class)) {
+            signatureUtilsMock.when(() -> SignatureUtils.isSignatureValid(any(), any(), any())).thenReturn(true);
+
+            assertTrue(workerpoolAuthorizationService.isSignedWithEnclaveChallenge(
+                    workerpoolAuthorization.getChainTaskId(),
+                    "mockSignature"));
+
+            signatureUtilsMock.verify(() ->
+                    SignatureUtils.isSignatureValid(
+                            any(byte[].class),
+                            any(Signature.class),
+                            eq(workerpoolAuthorization.getWorkerWallet())
+                    )
+            );
+        }
+    }
+
+    @Test
+    void shouldRejectSignatureNotSignedWithEnclaveChallenge() {
+        final WorkerpoolAuthorization workerpoolAuthorization = getWorkerpoolAuthorization();
+        final Map<String, WorkerpoolAuthorization> workerpoolAuthorizations = new HashMap<>();
+        workerpoolAuthorizations.put(workerpoolAuthorization.getChainTaskId(), workerpoolAuthorization);
+        ReflectionTestUtils.setField(workerpoolAuthorizationService, "workerpoolAuthorizations", workerpoolAuthorizations);
+
+        final Signature invalidSignature = SignatureUtils.signMessageHashAndGetSignature(
+                Arrays.toString(BytesUtils.stringToBytes("wrong-challenge")),
+                workerpoolAuthorization.getWorkerWallet()
+        );
+
+        assertFalse(workerpoolAuthorizationService.isSignedWithEnclaveChallenge(
+                workerpoolAuthorization.getChainTaskId(),
+                invalidSignature.toString()));
+    }
+
+    @Test
+    void shouldThrowExceptionWhenWorkerpoolAuthorizationNotFound() {
+        final Map<String, WorkerpoolAuthorization> workerpoolAuthorizations = new HashMap<>();
+        ReflectionTestUtils.setField(workerpoolAuthorizationService, "workerpoolAuthorizations", workerpoolAuthorizations);
+
```

Comment:
The `workerpoolAuthorizations` map is empty by default. These lines are not necessary.